### ISSUE TITLE
Valid zip file fails to extract with Zip headers not found exception

### ISSUE THREAD

**Issue by CalebFenton** [srikanth-lingala/zip4j#166]
I'm trying to open a malicious APK (which is just a ZIP) : https://www.virustotal.com/gui/file/3d442706c36549bb412684471746d4b6537751d8cdb6a40f4f18872a09fa7acd/detection

Attached: [48ff2a2b08d18d12d9110a3ad3c86ea0ba33487d.zip](https://github.com/srikanth-lingala/zip4j/files/4396035/48ff2a2b08d18d12d9110a3ad3c86ea0ba33487d.zip)

The file unzips fine. Only a warning about truncated comment.

```
$ unzip -t 3d4_failed.apk
Archive:  3d4_failed.apk

caution:  zipfile comment truncated
    testing: META-INF/MANIFEST.MF     OK
    testing: META-INF/CERT.SF         OK
    testing: META-INF/CERT.RSA        OK
    testing: .resouces_(manifest      OK
    testing: AndroidManifest.xml      OK
    testing: classes.dex              OK
    testing: res/drawable-hdpi-v11/ikbstwxtz.png   OK
    testing: res/drawable-hdpi-v4/ic_launcher.png   OK
    testing: res/drawable-hdpi-v4/ikbstwxtz.png   OK
    testing: res/drawable-mdpi-v11/ikbstwxtz.png   OK
    testing: res/drawable-mdpi-v4/ic_launcher.png   OK
    testing: res/drawable-mdpi-v4/ikbstwxtz.png   OK
    testing: res/drawable-xhdpi-v11/ikbstwxtz.png   OK
    testing: res/drawable-xhdpi-v4/ic_launcher.png   OK
    testing: res/drawable-xhdpi-v4/ikbstwxtz.png   OK
    testing: res/drawable-xxhdpi-v11/ikbstwxtz.png   OK
    testing: res/drawable-xxhdpi-v4/ic_launcher.png   OK
    testing: res/drawable-xxhdpi-v4/ikbstwxtz.png   OK
    testing: res/drawable-xxxhdpi-v11/ikbstwxtz.png   OK
    testing: res/drawable-xxxhdpi-v4/ikbstwxtz.png   OK
    testing: res/drawable/cpcqdnmlemk.xml   OK
    testing: res/layout-v17/dialog_web_view.xml   OK
    testing: res/layout/dialog_web_view.xml   OK
    testing: res/layout/fjtrduihiljh.xml   OK
    testing: res/layout/relative_layout.xml   OK
    testing: res/xml/wrqlcjxeg.xml    OK
    testing: resources.arsc           OK
No errors detected in compressed data of 3d4_failed.apk.
```

Zip4j 2.5.1 throws an exception:

```
Exception in thread "main" net.lingala.zip4j.exception.ZipException: Zip headers not found. Probably not a zip file or a corrupted zip file
	at net.lingala.zip4j.headers.HeaderReader.readAllHeaders(HeaderReader.java:78)
	at net.lingala.zip4j.ZipFile.readZipInfo(ZipFile.java:964)
	at net.lingala.zip4j.ZipFile.getFileHeader(ZipFile.java:582)
	at org.cf.apkfile.Main2.main(Main2.java:29)
Caused by: java.io.EOFException
	at java.io.RandomAccessFile.readFully(RandomAccessFile.java:438)
	at java.io.RandomAccessFile.readFully(RandomAccessFile.java:416)
	at net.lingala.zip4j.headers.HeaderReader.readEndOfCentralDirectoryRecord(HeaderReader.java:137)
	at net.lingala.zip4j.headers.HeaderReader.readAllHeaders(HeaderReader.java:74)
	... 3 more
```

To be fair, Java native libraries also have trouble opening the file:

```
Exception in thread "main" java.util.zip.ZipException: error in opening zip file
	at java.util.zip.ZipFile.open(Native Method)
	at java.util.zip.ZipFile.<init>(ZipFile.java:225)
	at java.util.zip.ZipFile.<init>(ZipFile.java:155)
	at java.util.zip.ZipFile.<init>(ZipFile.java:169)
	at org.cf.apkfile.Main2.main(Main2.java:26)
```

Oddly, running `zip -FF` (to fix the archive) works and creates another valid zip that Zip4j _also_ cannot open. I can upload that if you want, too.

**Comment by CalebFenton** [comment]
Sometimes malware authors create the files in weird, non standard ways or they intentionally munge the format to deter analysis.

**Comment by srikanth-lingala** [comment]
Issue fixed. I made changes to ignore reading zip comment when it cannot be read (the behaviour was to throw an exception earlier). I will include the fix in the next release.

**Comment by CalebFenton** [comment]
Wow, that was fast. Thank you kindly!

**Comment by srikanth-lingala** [comment]
Fix included in v2.5.2 released today.
